src/lib-components/overlays/Flash.vue

```diff
@@ -1,80 +1,40 @@
 <script setup lang="ts">
-import { Flash } from "@/composables/overlay"
-import { onMounted, ref } from "vue"
+import {
+  FlashType,
+  useAppFlashes,
+  loadWindowFlashes,
+} from "@/composables/useFlashes"
+import { onMounted } from "vue"
 
-// TODO: spk this might benefit from the composition api to avoid race conditions where a flash is requested before the component is mounted.
+const { flasher, flashes } = useAppFlashes()
 
-const flashes = ref<Flash[]>([])
-const flashTypeBorderClass = {
-  warning: "border-orange-500",
-  error: "border-red-500",
-  info: "border-blue-500",
-  success: "border-green-500",
-}
-
-const getFlashClass = (flash: Flash): string => {
-  const availableTypes = Object.keys(flashTypeBorderClass)
-  // default the flash type to "info" if no flash type or an unsupported flash.type is found
-  const type =
-    flash?.type && availableTypes.includes(flash.type)
-      ? (flash.type as "warning" | "error" | "info" | "success")
-      : "info"
-  return flashTypeBorderClass[type]
-}
-
-const remove = (flash: Flash): void => {
-  let index = 0
-  for (const f of flashes.value) {
-    if (flash.message === f.message) {
-      flashes.value.splice(index, 1)
-      return
-    }
-    index++
+const getFlashClass = (type: FlashType) => {
+  switch (type) {
+    case "warning":
+      return "border-orange-500"
+    case "error":
+      return "border-red-500"
+    case "info":
+      return "border-blue-500"
+    case "success":
+      return "border-green-500"
   }
 }
-const renderFlash = (flash: Flash): void => {
-  flashes.value.push(flash)
-  // Super simple flash implementation. This could get "smarter" by adding an
-  // id to the flash object, and then searching for the specific flash in the
-  // array and splicing, instead of simply doing a pop().
-  setTimeout(
-    (flashes: Flash[]) => {
-      flashes.pop()
-    },
-    10000,
-    flashes.value
-  )
-}
-
-const renderGenericError = (email: string): void => {
-  renderFlash({
-    type: "error",
-    message:
-      "Whoops! Something went wrong, please reach out to " +
-      `<a class="underline text-xy-blue" href="mailto:${email}">${email}</a>` +
-      " if the issue persists.",
-  })
-}
-
 onMounted(() => {
+  // NOTE: (spk) here for backwards compatibility
```

Comment:
I'm down to remove this here, and migrate archive/tax/portal rather than making it backwards compatible. It's not that large of a task, and it helps us move to a world without mitt.

Reply:
That sounds good to me as well.  I wasn't sure of everyone's appetite for breaking changes, but agreed this should be a pretty quick and easy update.